Wikidata.js

```diff
@@ -0,0 +1,562 @@
+	url = url.replace(/#.*/, '');
+	var qposition = url.indexOf('Q');
+	var qnumber = url.substr(qposition);
+	var xmlUrl = ZU.xpathText(doc, '//link[@rel="alternate" and @type="application/rdf+xml"]/@href');
+	ZU.doGet(xmlUrl, function(data) {
+		var parser = new DOMParser();
+		var xml = parser.parseFromString(data, "application/xml");
+		
+		var item = new Zotero.Item(type);
+		var nodes = ZU.xpath(xml, '//rdf:Description', namespaces);
+		for (var i=0; i<nodes.length; i++) {
+			var rdfabout = nodes[i].getAttribute("rdf:about");
+			if (rdfabout) {
+				var id = rdfabout.substr(rdfabout.length-qnumber.length);
+				if (id==qnumber) {
+					for (var prop in nodes[i].childNodes) {
+						var propstatement = nodes[i].childNodes[prop];
+						var tagname = propstatement.tagName;
+						if (tagname && mapping[tagname]) {
+							var zprop = mapping[tagname];
+							var value = propstatement.textContent;
+							var resource = propstatement.getAttribute("rdf:resource");
+							if (!value && resource) {
+								//Z.debug("Internal look up resource: " + resource);
+								value = ZU.xpathText(xml, '//rdf:Description[@rdf:about="'+resource+'"]/rdfs:label[contains(@xml:lang, "en")][1]', namespaces)
+									|| ZU.xpathText(xml, '//rdf:Description[@rdf:about="'+resource+'"]/rdfs:label[1]', namespaces);
+								//Z.debug(value);
+							}
+							if (zprop=="creator") {
+								var func = creatorMapping[tagname] || 'contributor';
+								item.creators.push(ZU.cleanAuthor(value, func));
+							} else if (item[zprop]) {
+								item[zprop] += ', ' + value;
+							} else {
+								item[zprop] = value;
+							}
+						}
+					}
+				}
+			}
+		}
+		if (item.title && item.subtitle) {
+			item.title += ': ' + item.subtitle;
+			delete item.subtitle;
+		}
+		if (item.tagString) {
+			var tags = item.tagString.split(', ');
+			for (var j=0; j<tags.length; j++) {
+				item.tags.push(tags[j]);
+			}
+			delete item.tagString;
+		}
+
+		item.complete();
+	});
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://www.wikidata.org/wiki/Q30000000",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "The Synergistic Activity of Thyroid Transcription Factor 1 and Pax 8 Relies on the Promoter/Enhancer Interplay",
+				"creators": [
+					{
+						"firstName": "Stefania",
+						"lastName": "Miccadei",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Enrico",
+						"lastName": "Zammarchi",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Rossana De",
+						"lastName": "Leo",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Donato",
+						"lastName": "Civitareale",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Pier Giorgio",
+						"lastName": "Natali",
+						"creatorType": "author"
+					}
+				],
+				"date": "2002-04-01T00:00:00Z",
+				"DOI": "10.1210/MEND.16.4.0808, 10.1210/ME.16.4.837",
```

Comment:
Well, if it breaks something in Zotero, and there's no fix/ workaround, by all means drop one of them. Let's not block deployment of this useful script, nor let "perfect be the enemy of good enough".

How will you be sure to keep the best one? ;-)